packages/fcl-wc/src/service.js

```diff
@@ -0,0 +1,170 @@
+import QRCodeModal from "@walletconnect/qrcode-modal"
+
+export const makeServicePlugin = client => ({
+  name: "fcl-service-walletconnect",
+  type: "discovery-service",
+  services: makeWcServices(client),
+})
+
+const makeServiceStrategy = client => {
+  return ({service, body, opts}) => {
+    return new Promise(async (resolve, reject) => {
+      if (typeof client === "undefined") {
+        throw new Error("WalletConnect is not initialized")
+      }
+      let session
+      const onResponse = resp => {
+        try {
+          if (typeof resp !== "object") return
+
+          switch (resp.status) {
+            case "APPROVED":
+              resolve(resp.data)
+              break
+
+            case "DECLINED":
+              reject(`Declined: ${resp.reason || "No reason supplied"}`)
+              break
+
+            case "REDIRECT":
+              resolve(resp)
+              break
+
+            default:
+              reject(`Declined: No reason supplied`)
+              break
+          }
+        } catch (error) {
+          console.error("execWcRPC onResponse error", error)
+          throw error
+        }
+      }
+
+      const onClose = () => {
+        reject(`Declined: Externally Halted`)
+      }
+
+      if (client.session.length) {
+        const lastKeyIndex = client.session.keys.length - 1
+        session = client.session.get(client.session.keys[lastKeyIndex])
```

Comment:
What if `client.session.keys` doesn't exist or length is zero? You could use JS's new `at()` method to look more gracefully.

Reply:
at this point there is a session/s, and session.keys lists all session topic strings (length > 0)
new js methods LFG